src/hotspot/share/gc/g1/heapRegion.hpp

```diff
@@ -165,16 +165,12 @@ class HeapRegion : public CHeapObj<mtGC> {
 
   HeapWord* initialize_threshold();
   HeapWord* cross_threshold(HeapWord* start, HeapWord* end);
-  // Update heap region to be consistent after Full GC compaction.
-  void reset_humongous_during_compaction() {
-    assert(is_humongous(),
-           "should only be called for humongous regions");
 
-    zero_marked_bytes();
-    init_top_at_mark_start();
-  }
-  // Update heap region to be consistent after Full GC compaction.
-  void complete_compaction();
+  // Update heap region to be consistent after Full GC compaction for non-pinned
+  // regions.
+  void non_pinned_complete_compaction();
+  void pinned_complete_compaction();
+  void complete_compaction_common();
```

Comment:
What do you think about using these names instead:
```suggestion
  void complete_compaction();
  void reset_pinned();
  void reset_after_full();
```

An other alternative would also to be to use `reset_after_compaction()` instead of `complete_compaction()`.